src/sdk/accountspace_nb.js

```diff
@@ -247,6 +247,136 @@ class AccountSpaceNB {
+
+        account_util._check_if_requesting_account_is_root_account(action, requesting_account, { username: params.username });
+        account_util._check_if_account_exists(action, username);
+
+        const requested_account = system_store.get_account_by_email(username);
+        account_util._check_if_requested_account_is_root_account_or_IAM_user(action, requesting_account, requested_account);
+        account_util._check_if_requested_is_owned_by_root_account(action, requesting_account, requested_account);
+
+        const existing_tags = requested_account.tagging || [];
+
+        const tag_keys_set = new Set(params.tag_keys);
+        const updated_tags = existing_tags.filter(tag => !tag_keys_set.has(tag.key));
+
+        await system_store.make_changes({
+            update: {
+                accounts: [{
+                    _id: requested_account._id,
+                    $set: { tagging: updated_tags }
+                }]
+            }
+        });
+
+        dbg.log1('AccountSpaceNB.untag_user: successfully removed', params.tag_keys.length, 'tags from user', params.username);
+    }
+
+    async list_user_tags(params, account_sdk) {
+        const action = IAM_ACTIONS.LIST_USER_TAGS;
+        const requesting_account = system_store.get_account_by_email(account_sdk.requesting_account.email);
+        const username = account_util.populate_username(params.username, requesting_account.name.unwrap());
+
+        account_util._check_if_requesting_account_is_root_account(action, requesting_account, { username: params.username });
+        account_util._check_if_account_exists(action, username);
+
+        const requested_account = system_store.get_account_by_email(username);
+        account_util._check_if_requested_account_is_root_account_or_IAM_user(action, requesting_account, requested_account);
+        account_util._check_if_requested_is_owned_by_root_account(action, requesting_account, requested_account);
+
+        const all_tags = requested_account.tagging || [];
+
+        const sorted_tags = all_tags.sort((a, b) => a.key.localeCompare(b.key));
+
+        let start_index = 0;
+        if (params.marker) {
+            start_index = sorted_tags.findIndex(tag => tag.key === params.marker);
+            if (start_index === -1) {
+                start_index = 0;
+            } else {
+                start_index += 1;
+            }
+        }
+
+        const max_items = params.max_items;
+        const end_index = start_index + max_items;
+        const paginated_tags = sorted_tags.slice(start_index, end_index);
+
+        const tags = paginated_tags.map(tag => ({
+            member: {
+                Key: tag.key,
+                Value: tag.value
+            }
+        }));
+
+        const is_truncated = end_index < sorted_tags.length;
+        const marker = is_truncated && paginated_tags.length > 0 ? paginated_tags[paginated_tags.length - 1].key : undefined;
```

Comment:
Could you move it to a separate function?
We didn't implement pagination so far, and if you move it to a function it can be reusable.